common/src/glob.rs

```diff
@@ -0,0 +1,64 @@
+//! Contains `globset::Glob` wrapper functions used for filtering
+use std::{fmt, str::FromStr};
+/// A `globset::Glob` that creates its `globset::GlobMatcher` when its created, so it doesn't need
+/// to be compiled when the filter functions `TestFilter` functions are called.
+#[derive(Debug, Clone)]
+pub struct GlobMatcher {
+    /// The parsed glob
+    pub glob: globset::Glob,
+    /// The compiled `glob`
+    pub matcher: globset::GlobMatcher,
+}
+
+// === impl GlobMatcher ===
+
+impl GlobMatcher {
+    /// Tests whether the given path matches this pattern or not.
+    ///
+    /// The glob `./test/*` won't match absolute paths like `test/Contract.sol`, which is common
+    /// format here, so we also handle this case here
+    pub fn is_match(&self, path: &str) -> bool {
```

Comment:
Ideally to match `/private/var/folders/test/Foo.sol` (this is how it appears in the tests) one wouldn't need `*/test/**` and could instead use `./test/*`. I don't have insight into how paths are passed around in Foundry (are they always absolute?), but I could appreciate clarification on this point.

For example, this is the cause of the failing test: 
```
        let file = Path::new("/private/var/folders/test/Foo.sol");
        assert!(!SkipBuildFilter::Custom("*/test/**".to_string()).is_match(file)); // passes 
        let file = Path::new("script/Contract.sol");
        assert!(!SkipBuildFilter::Custom("*/script/**".to_string()).is_match(file)); // fails
```

Reply:
> have insight into how paths are passed around in Foundry (are they always absolute?)

they're relative to the configured root when dealing solc, outside of solc they always should be absolute, so foundry should always have absolute paths